### DISCUSSION TITLE
How to convert a BitVec type in a constraint expression to an Int type

### DISCUSSION THREAD

**Discussion by DQP2375** [Z3Prover/z3#5999] [Q&A]
Hello Developers! I'm recently learning to do some more work with z3 (python) and I now want to convert the BitVec type in a constraint expression obtained from a symbolic execution to an Int type, how do I do that?

**Comment by NikolajBjorner** [comment]
There are built in functions to convert bit-vectors to integers. There are signed and unsigned versions. 
Look for the function BV2Int in the python API.

**Comment by DQP2375** [comment]
Thanks for your answer! I know this function, maybe I didn't express it clearly, and my problem doesn't seem to be able to use this method.
1. I directly obtained the constraint expression of the program through the symbolic execution tool and then converted it into the standard input of z3, **getting an expression**! The symbol variable in this expression is defined as BitVec, but we need to convert all BitVec types in the whole expression to Int type. BV2Int seems to only be able to transform variables but not expressions. So I want to help you.
2. I still have a question, as written in the code below, we calculate x*x<0, which will never be true in mathematical logic, but I don't know why the result of BitVec type calculation exists, so we need to put Converting the BitVec type to an Int to avoid such a situation, would adding a constraint otherwise solve the problem? Or there are other better solutions
x = BitVec('x', 32)
s = Solver()
exper = x * x < 0
s.add(exper)
if s.check() == sat:
    solution = s.model().eval(x)
    print(solution)
else:
    print('no any')
Forgive me for being a rookie! Thanks in advance!
